events/src/utils.py

```diff
@@ -313,6 +314,40 @@ def plugin_webhook(message):
     webhooks_pool.send(message["event"], message["ts"], message_json)
 
 
+class NamedTempOpen(object):
+    """
+        This class is used to create a temporary file which is then written to with contents.
+        The temp file is then persisted with the give name by calling os.rename().
+        This class is used to avoid the data loss or truncation in case of multiple processes
+        writing to the same file without the use of fcntl locks.
+
+        The temporary file is created in the cwd of the process.
+    """
+
+    def __init__(self, filename, open_mode, *args, **kwagrs) -> None:
+        self.filename = filename
+        self.open_mode = open_mode
+        self.open_args = args
+        self.open_kwargs = kwagrs
+        self.fileobj = None
+
+    def __enter__(self):
+        tfile = NamedTemporaryFile(mode=self.open_mode,
+                                   delete=False,
+                                   dir=".",
+                                   *self.open_args,
+                                   **self.open_kwargs)
+
+        if tfile is None:
```

Comment:
Wouldn't it raise an exception instead of returning `None` ?

Reply:
It is supposed to raise an exception in case things don't go in accordance. Returning `None` would further complicate things as we only ever use this class with `with` keyword.

Reply:
If it raises exceptions in case of error, if everything goes well, it will always return an object. Why do we need to check if it's `None` ?

Reply:
Removed the check. Wrapped the temp file creation inside `try/except` blocks to raise the base exception.